composite/market-closure/src/checks/schedule.ts

```diff
@@ -6,11 +6,11 @@ const customParams = {
   schedule: false,
 }
 
-export const isMarketClosedFactory = (input: AdapterRequest): (() => Promise<boolean>) => {
+export const isMarketClosedFactory = (input: AdapterRequest): Promise<boolean> => {
   const validator = new Validator(input, customParams)
   if (validator.error) throw validator.error
   const schedule = validator.validated.data.schedule || {}
-  return async () => isMarketClosed(schedule as Schedule)
+  return new Promise<boolean>((resolve) => resolve(isMarketClosed(schedule as Schedule)))
```

Comment:
- You want to return a promise to be consistent with some Types I guess right?
- An easier way to make it return a Promise, could be:
    -  Make the function `async`. That automatically returns a Promise
    - Or, `return Promise.resolve(isMarketClosed(...))`

Also, it seems that you could merge `isMarketClosedFactory` and `isMarketClosed` into one single function `isMarketClosed`.

Reply:
Ah, of course - I think that's what I intended, don't know how I ended up with what I have here hahah